Transient "cannot find module" when running gulp test

bradfrizzell and I have been experiencing frequent errors like the following when running "gulp test", both locally on our machines as well as on Travis. Restarting the test, or rebuilding and then re-running the test, generally fix the issue.
The module that can't be found seems to vary.

```
  {
    "message": "Uncaught Error: Cannot find module '/home/travis/build/ampproject/amphtml/test/integration/test-3p-frame.js'\nat /tmp/node_modules/browser-pack/_prelude.js:1:0 <- /tmp/d65e3526f9c22abf2a123e2513182be5.browserify:1:196\n\nError: Cannot find module '/home/travis/build/ampproject/amphtml/test/integration/test-3p-frame.js'\n    at s (/tmp/node_modules/browser-pack/_prelude.js:1:0 <- /tmp/d65e3526f9c22abf2a123e2513182be5.browserify:1:156)\n    at s (/tmp/node_modules/browser-pack/_prelude.js:1:0 <- /tmp/d65e3526f9c22abf2a123e2513182be5.browserify:1:122)\n    at test/integration/test-3p-frame.js:1:34",
    "str": "Uncaught Error: Cannot find module '/home/travis/build/ampproject/amphtml/test/integration/test-3p-frame.js'\nat /tmp/node_modules/browser-pack/_prelude.js:1:0 <- /tmp/d65e3526f9c22abf2a123e2513182be5.browserify:1:196\n\nError: Cannot find module '/home/travis/build/ampproject/amphtml/test/integration/test-3p-frame.js'\n    at s (/tmp/node_modules/browser-pack/_prelude.js:1:0 <- /tmp/d65e3526f9c22abf2a123e2513182be5.browserify:1:156)\n    at s (/tmp/node_modules/browser-pack/_prelude.js:1:0 <- /tmp/d65e3526f9c22abf2a123e2513182be5.browserify:1:122)\n    at test/integration/test-3p-frame.js:1:34"
  }
```

@rsimha Here is an example. I ran the following commands:
```
cd ~/workspace
mkdir unskip_tests
cd unskip_tests
git clone git@github.com:jonkeller/amphtml.git
cd amphtml
git remote add upstream git@github.com:ampproject/amphtml.git
git fetch upstream
git checkout -t -b upstream_master upstream/master
git remote -v
git branch
git checkout upstream_master
git pull upstream --ff-only
git checkout master
git branch -u upstream/master
git pull upstream master
yarn
gulp clean
gulp build
git checkout -b unskip_tests_jonkeller
gulp test --files=extensions/amp-analytics/0.1/test/test-amp-analytics.js --nobuild
```

...and immediately got:
```
START:
Chrome 65.0.3325 (Mac OS X 10.13.3) ERROR
  {
    "message": "Uncaught Error: Cannot find module '/Users/jonkeller/workspace/unskip_tests/amphtml/extensions/amp-analytics/0.1/test/test-amp-analytics.js'\nat /var/folders/93/v772f58d0m5gpp5nq6l1yz2800dbvk/T/node_modules/browser-pack/_prelude.js:1:0 <- /var/folders/93/v772f58d0m5gpp5nq6l1yz2800dbvk/T/7e1556c39cab1e7aaca343091b581d30.browserify:1:196\n\nError: Cannot find module '/Users/jonkeller/workspace/unskip_tests/amphtml/extensions/amp-analytics/0.1/test/test-amp-analytics.js'\n    at s (/var/folders/93/v772f58d0m5gpp5nq6l1yz2800dbvk/T/node_modules/browser-pack/_prelude.js:1:0 <- /var/folders/93/v772f58d0m5gpp5nq6l1yz2800dbvk/T/7e1556c39cab1e7aaca343091b581d30.browserify:1:156)\n    at s (/var/folders/93/v772f58d0m5gpp5nq6l1yz2800dbvk/T/node_modules/browser-pack/_prelude.js:1:0 <- /var/folders/93/v772f58d0m5gpp5nq6l1yz2800dbvk/T/7e1556c39cab1e7aaca343091b581d30.browserify:1:122)\n    at extensions/amp-analytics/0.1/test/test-amp-analytics.js:1:34",
    "str": "Uncaught Error: Cannot find module '/Users/jonkeller/workspace/unskip_tests/amphtml/extensions/amp-analytics/0.1/test/test-amp-analytics.js'\nat /var/folders/93/v772f58d0m5gpp5nq6l1yz2800dbvk/T/node_modules/browser-pack/_prelude.js:1:0 <- /var/folders/93/v772f58d0m5gpp5nq6l1yz2800dbvk/T/7e1556c39cab1e7aaca343091b581d30.browserify:1:196\n\nError: Cannot find module '/Users/jonkeller/workspace/unskip_tests/amphtml/extensions/amp-analytics/0.1/test/test-amp-analytics.js'\n    at s (/var/folders/93/v772f58d0m5gpp5nq6l1yz2800dbvk/T/node_modules/browser-pack/_prelude.js:1:0 <- /var/folders/93/v772f58d0m5gpp5nq6l1yz2800dbvk/T/7e1556c39cab1e7aaca343091b581d30.browserify:1:156)\n    at s (/var/folders/93/v772f58d0m5gpp5nq6l1yz2800dbvk/T/node_modules/browser-pack/_prelude.js:1:0 <- /var/folders/93/v772f58d0m5gpp5nq6l1yz2800dbvk/T/7e1556c39cab1e7aaca343091b581d30.browserify:1:122)\n    at extensions/amp-analytics/0.1/test/test-amp-analytics.js:1:34"
  }
```

I realize I used --nobuild in executing the test. But I had just done "gulp build" immediately prior. Shouldn't this work? It always used to.

Update: Then I ran the same "gulp test" command, without the "--nobuild", and got the same error:

```
gulp test --files=extensions/amp-analytics/0.1/test/test-amp-analytics.js
[15:09:48] Using gulpfile ~/workspace/unskip_tests/amphtml/gulpfile.js
[15:09:48] Starting 'update-packages'...
[15:09:48] All packages in node_modules are up to date.
[15:09:48] Finished 'update-packages' after 663 ms
[15:09:48] Starting 'build'...
[15:09:48] Running build. Press Ctrl + C to cancel...
[15:09:48] To skip building during future test runs, use --nobuild with your gulp test command.
[15:09:48] Building the runtime for local testing with the prod AMP config.
[15:09:48] ⤷ Use --config={canary|prod} with your gulp build command to specify which config to apply.
[15:09:48] Building all AMP extensions.
[15:09:48] ⤷ Use --noextensions to skip building extensions.
[15:09:48] ⤷ Use --extensions=amp-foo,amp-bar to choose which extensions to build.
[15:09:48] ⤷ Use --extensions=minimal_set to build just the extensions needed to load article.amp.html.
[15:09:49] Recompiled CSS in amp.css (505 ms)
[15:09:49] Recompiled CSS in amp-access (560 ms)
[15:09:49] Recompiled CSS in amp-access-scroll (559 ms)
[15:09:49] Recompiled CSS in amp-ad (558 ms)
[15:09:49] Recompiled CSS in amp-apester-media (558 ms)
[15:09:49] Recompiled CSS in amp-byside-content (555 ms)
[15:09:49] Recompiled CSS in amp-consent (550 ms)
[15:09:49] Recompiled CSS in amp-document-recommendations (549 ms)
[15:09:49] Recompiled CSS in amp-fit-text (548 ms)
[15:09:49] Recompiled CSS in amp-form (548 ms)
[15:09:49] Recompiled CSS in amp-fx-flying-carpet (547 ms)
[15:09:49] Recompiled CSS in amp-gwd-animation (546 ms)
[15:09:49] Recompiled CSS in amp-image-lightbox (545 ms)
[15:09:49] Recompiled CSS in amp-instagram (543 ms)
[15:09:49] Recompiled CSS in amp-lightbox (542 ms)
[15:09:49] Recompiled CSS in amp-live-list (543 ms)
[15:09:49] Recompiled CSS in amp-mathml (543 ms)
[15:09:49] Recompiled CSS in amp-pinterest (542 ms)
[15:09:49] Recompiled CSS in amp-sidebar (539 ms)
[15:09:49] Recompiled CSS in amp-story-auto-ads (512 ms)
[15:09:49] Recompiled CSS in amp-selector (512 ms)
[15:09:49] Recompiled CSS in amp-web-push (511 ms)
[15:09:49] Recompiled CSS in amp-image-viewer (509 ms)
[15:09:49] Recompiled CSS in amp-user-notification (506 ms)
[15:09:49] Recompiled CSS in amp-viz-vega (506 ms)
[15:09:49] Recompiled CSS in amp-access-laterpay (572 ms)
[15:09:49] Recompiled CSS in amp-app-banner (568 ms)
[15:09:49] Recompiled CSS in amp-carousel (566 ms)
[15:09:49] Recompiled CSS in amp-playbuzz (547 ms)
[15:09:49] Recompiled CSS in amp-sticky-ad (542 ms)
[15:09:49] Recompiled CSS in amp-subscriptions (515 ms)
[15:09:49] Recompiled CSS in amp-lightbox-gallery (555 ms)
[15:09:50] Recompiled CSS in amp-date-picker (712 ms)
[15:09:50] Recompiled CSS in amp-social-share (740 ms)
[15:09:50] Recompiled CSS in amp-story (1.186 s)
[15:09:50] Processed 3p/frame.max.html (025 ms)
[15:09:50] Processed 3p/nameframe.max.html (024 ms)
[15:09:52] Compiled examiner.js (1.790 s)
[15:09:52] Compiled kill.js (1.794 s)
[15:10:01] Compiled core.js (11.068 s)
[15:10:24] Compiled shell.js (33.840 s)
[15:10:26] Compiled ampcontext-lib.js (35.995 s)
[15:10:26] Compiled integration.js (36.022 s)
[15:10:26] Compiled iframe-transport-client-lib.js (36.080 s)
[15:10:26] No configs found in dist.3p/current/integration.js
[15:10:26] Compiled polyfills.js (36.327 s)
[15:10:26] Enabled local development mode in dist.3p/current/integration.js
[15:10:27] Wrote prod AMP config to dist.3p/current/integration.js
[15:10:27] Compiled web-worker.js (36.734 s)
[15:10:29] Compiled amp-viewer-host.js (39.120 s)
[15:10:30] Compiled amp.js (39.552 s)
[15:10:30] Compiled amp-shadow.js (39.544 s)
[15:10:30] Compiled amp-inabox.js (39.573 s)
[15:10:30] Compiled amp-analytics.js (39.564 s)
[15:10:30] No configs found in dist/amp.js
[15:10:30] Compiled install-alp.js (39.958 s)
[15:10:30] Compiled inabox-host.js (39.891 s)
[15:10:30] Enabled local development mode in dist/amp.js
[15:10:30] Wrote prod AMP config to dist/amp.js
[15:10:30] Running npx bundlesize -f "./dist/amp.js" -s "333.28KB"...

 PASS  ./dist/amp.js: 333.27KB < maxSize 333.28KB (gzip)

[15:10:31] Finished 'build' after 43 s
[15:10:31] Starting 'test'...
[15:10:31] Run gulp help to see a list of all test flags.
[15:10:31] ⤷ Use --nohelp to silence these messages.)
[15:10:31] ⤷ Use --headless to run tests in a headless Chrome window.
[15:10:31] Running tests against unminified code.
[15:10:31] --files: Running tests in the file(s): extensions/amp-analytics/0.1/test/test-amp-analytics.js
[15:10:31] Setting the runtime's AMP config to prod
[15:10:31] Removed existing config from dist/amp.js
[15:10:31] Enabled local development mode in dist/amp.js
[15:10:32] Wrote prod AMP config to dist/amp.js
[15:10:32] Webserver started at http://localhost:31862
[15:10:32] Started test responses server on localhost:31862
[15:10:32] Running test. Press Ctrl + C to cancel...
Running tests locally...

START:
Chrome 65.0.3325 (Mac OS X 10.13.3) ERROR
  {
    "message": "Uncaught Error: Cannot find module '/Users/jonkeller/workspace/unskip_tests/amphtml/extensions/amp-analytics/0.1/test/test-amp-analytics.js'\nat /var/folders/93/v772f58d0m5gpp5nq6l1yz2800dbvk/T/node_modules/browser-pack/_prelude.js:1:0 <- /var/folders/93/v772f58d0m5gpp5nq6l1yz2800dbvk/T/79a9552678943b8cd8d7cc5b18b319e2.browserify:1:196\n\nError: Cannot find module '/Users/jonkeller/workspace/unskip_tests/amphtml/extensions/amp-analytics/0.1/test/test-amp-analytics.js'\n    at s (/var/folders/93/v772f58d0m5gpp5nq6l1yz2800dbvk/T/node_modules/browser-pack/_prelude.js:1:0 <- /var/folders/93/v772f58d0m5gpp5nq6l1yz2800dbvk/T/79a9552678943b8cd8d7cc5b18b319e2.browserify:1:156)\n    at s (/var/folders/93/v772f58d0m5gpp5nq6l1yz2800dbvk/T/node_modules/browser-pack/_prelude.js:1:0 <- /var/folders/93/v772f58d0m5gpp5nq6l1yz2800dbvk/T/79a9552678943b8cd8d7cc5b18b319e2.browserify:1:122)\n    at extensions/amp-analytics/0.1/test/test-amp-analytics.js:1:34",
    "str": "Uncaught Error: Cannot find module '/Users/jonkeller/workspace/unskip_tests/amphtml/extensions/amp-analytics/0.1/test/test-amp-analytics.js'\nat /var/folders/93/v772f58d0m5gpp5nq6l1yz2800dbvk/T/node_modules/browser-pack/_prelude.js:1:0 <- /var/folders/93/v772f58d0m5gpp5nq6l1yz2800dbvk/T/79a9552678943b8cd8d7cc5b18b319e2.browserify:1:196\n\nError: Cannot find module '/Users/jonkeller/workspace/unskip_tests/amphtml/extensions/amp-analytics/0.1/test/test-amp-analytics.js'\n    at s (/var/folders/93/v772f58d0m5gpp5nq6l1yz2800dbvk/T/node_modules/browser-pack/_prelude.js:1:0 <- /var/folders/93/v772f58d0m5gpp5nq6l1yz2800dbvk/T/79a9552678943b8cd8d7cc5b18b319e2.browserify:1:156)\n    at s (/var/folders/93/v772f58d0m5gpp5nq6l1yz2800dbvk/T/node_modules/browser-pack/_prelude.js:1:0 <- /var/folders/93/v772f58d0m5gpp5nq6l1yz2800dbvk/T/79a9552678943b8cd8d7cc5b18b319e2.browserify:1:122)\n    at extensions/amp-analytics/0.1/test/test-amp-analytics.js:1:34"
  }

Finished in 0.886 secs / 0 secs @ 15:10:58 GMT-0400 (EDT)

SUMMARY:
● 0 tests completed
[15:10:59] Shutting down test responses server on localhost:31862
[15:10:59] ERROR: Karma test failed with exit code 1
 ✘ jonkeller@jonkeller-macbookpro  ~/workspace/unskip_tests/amphtml   unskip_tests_jonkeller ● 
```

--- a/build-system/tasks/karma.conf.js
+++ b/build-system/tasks/karma.conf.js
@@ -35,16 +35,17 @@ module.exports = {
   ],
 
   preprocessors: {
-    'test/fixtures/*.html': ['html2js'],
-    'test/**/*.js': ['browserify'],
-    'ads/**/test/test-*.js': ['browserify'],
-    'extensions/**/test/**/*.js': ['browserify'],
-    'testing/**/*.js': ['browserify'],
+    './test/fixtures/*.html': ['html2js'],
+    './test/**/*.js': ['browserify'],
+    './ads/**/test/test-*.js': ['browserify'],
+    './extensions/**/test/**/*.js': ['browserify'],
+    './testing/**/*.js': ['browserify'],
   },
 
   browserify: {
     watch: true,
     debug: true,
+    basedir: __dirname + '/../../',
     transform: [
       ['babelify', {compact: false}],
     ],
@@ -128,7 +129,7 @@ module.exports = {
     },
     Chrome_no_extensions_headless: {
       base: 'ChromeHeadless',
-      flags: COMMON_CHROME_FLAGS,
+      flags: ['--no-sandbox'].concat(COMMON_CHROME_FLAGS),
     },
     // SauceLabs configurations.
     // New configurations can be created here:

--- a/build-system/tasks/runtime-test.js
+++ b/build-system/tasks/runtime-test.js
@@ -169,7 +169,7 @@ function printArgvMessages() {
     log(green('Run'), cyan('gulp help'),
         green('to see a list of all test flags.'));
     log(green('⤷ Use'), cyan('--nohelp'),
-        green('to silence these messages.)'));
+        green('to silence these messages.'));
     if (!argv.unit && !argv.integration && !argv.files && !argv.a4a) {
       log(green('Running all tests.'));
       log(green('⤷ Use'), cyan('--unit'), green('or'), cyan('--integration'),

